Add TelnetServer man page and add sections to Help/man

larb0b opened the pull request.

bugaevc commented:
You should also add it to the list in `man(1)`; the problem with that is that it's an ordered list, so even if you write:

```markdown
1. Foo
2. Bar
8. Baz
```

it will render like

1. Foo
2. Bar
8. Baz

bugaevc commented on Base/usr/share/man/man8/TelnetServer.md:
````diff
@@ -0,0 +1,25 @@
+## Synopsis
+
+```**sh
+$ TelnetServer [options]
````
The `TelnetServer` is supposed to be run as root (unless you're using a custom port), correct? Then let's make the prompt a hash sign instead of a dollar.

bugaevc commented on Base/usr/share/man/man8/TelnetServer.md:
```diff
@@ -0,0 +1,25 @@
+
+## Options
+
+* `-p`: Choose different port for TelnetServer to attach to.
```
So should "TelnetServer" be monospaced or not? I'd say no, but what's important is that we pick one way and stick to it :)

bugaevc commented on Base/usr/share/man/man8/TelnetServer.md:
```diff
@@ -0,0 +1,25 @@
+## Options
+
+* `-p`: Choose different port for TelnetServer to attach to.
+* `-c`: Choose different program for TelnetServer to run on connection.
```
Can it accept more args to the command, ex. `TelnetServer -c /bin/foo argv1 argv2`? Let's mention it either way.

bugaevc commented on Base/usr/share/man/man8/TelnetServer.md:
````diff
@@ -0,0 +1,25 @@
+## Examples
+
+```sh
+$ TelnetServer -p 24 -c /usr/bin/nyancat
````
Here, too, the prompt should be a `#`

awesomekling commented on Applications/Help/ManualModel.cpp:
```diff
@@ -6,7 +6,8 @@
     { "1", "Command-line programs" },
-    { "2", "System calls" }
+    { "2", "System calls" },
+    { "8", "System administration commands and daemons" }
```
This is a long name.. does it look OK in the Help app?

larb0b replied:
It's cut off but you can expand the view of course. That's the "standard" name kind of. Is there a different one you'd prefer?

larb0b commented:
> 
> 
> You should also add it to the list in `man(1)`; the problem with that is that it's an ordered list, so even if you write:
> 
> ```gfm
> 1. Foo
> 2. Bar
> 8. Baz
> ```
> 
> it will render like
> 
>     1. Foo
> 
>     2. Bar
> 
>     3. Baz

By this do you mean that TelnetServer should be in man(1) and man(8)? On other systems something like this would be in 8 or 1M, where both are `System administration commands and daemons` or something similar.

bugaevc commented:
No, I mean the `man(1)` page (man page of the `man` program itself) also has a list of sections which you should update.

larb0b commented:
> 
> 
> No, I mean the `man(1)` page (man page of the `man` program itself) also has a list of sections which you should update.

Hm... should I just fill in all of them even if some are currently empty?

e.g.

1. Command-line programs
2. System calls
3. Library functions
4. Special files
5. File formats
6. Games
7. Miscellanea
8. System administration commands and daemons

bugaevc commented:
That's what I'm not sure about, and I won't complain either way :) but if you addjust the 8th section, make sure, somehow, that it gets number 8, not 3.

> 3. Library functions

I guess the name should be just "libraries", because Serenity libraries are mostly about C++ classes, not functions. And that's what I'm also unsure about — should section 3 pages explain whole libraries, classes or individual methods? ... classes, I guess.

> 8. System administration commands and daemons

Yeah, you really need to think of a shorter name :)

larb0b commented:
Alright, I've renamed 3 to "Libraries" and 8 to "Sysadmin tools". Also changed the note in man.md to say that the section titles are subject to change.

bugaevc commented:
LGTM now :)

awesomekling commented:
Very nice!
